internal/generate/olm-catalog/package_manifest.go

```diff
@@ -96,41 +95,81 @@ func (g pkgGenerator) Generate() error {
 // generate either reads an existing package manifest or creates a new
 // manifest and modifies it based on values set in s.
 func (g pkgGenerator) generate() (map[string][]byte, error) {
+	fileName, pkg, err := g.buildPackageManifest()
+	if err != nil {
+		return nil, err
+	}
+
+	g.setChannels(&pkg)
+	g.sortChannelsByName(&pkg)
+
+	if err := validatePackageManifest(&pkg); err != nil {
+		return nil, fmt.Errorf("error validating package manifest %s: %v", pkg.PackageName, err)
+	}
+
+	b, err := yaml.Marshal(pkg)
+	if err != nil {
+		return nil, err
+	}
+	fileMap := map[string][]byte{
+		fileName: b,
+	}
+	return fileMap, nil
+}
+
+// buildPackageManifest will build the registry.PackageManifest which is the structure used by OLM
+// by reading the files found in the path.
+func (g pkgGenerator) buildPackageManifest() (string, registry.PackageManifest, error) {
+	pkg := registry.PackageManifest{}
 	fileName := getPkgFileName(g.OperatorName)
 	path := filepath.Join(g.Inputs[ManifestsDirKey], fileName)
-	pkg := registry.PackageManifest{}
 	if _, err := os.Stat(path); err == nil {
 		b, err := ioutil.ReadFile(path)
 		if err != nil {
-			return nil, fmt.Errorf("failed to read package manifest %s: %v", path, err)
+			return "", pkg, fmt.Errorf("failed to read package manifest %s: %v", path, err)
 		}
 		if err = yaml.Unmarshal(b, &pkg); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal package manifest %s: %v", path, err)
-		}
-		if err := registryutil.ValidatePackageManifest(&pkg); err != nil {
-			return nil, fmt.Errorf("error validating package manifest %s: %v", pkg.PackageName, err)
+			return "", pkg, fmt.Errorf("failed to unmarshal package manifest %s: %v", path, err)
 		}
 	} else if os.IsNotExist(err) {
 		pkg = newPackageManifest(g.OperatorName, g.channel, g.csvVersion)
 	} else {
-		return nil, fmt.Errorf("error reading package manifest %s: %v", path, err)
+		return "", pkg, fmt.Errorf("error reading package manifest %s: %v", path, err)
 	}
+	return fileName, pkg, nil
+}
 
-	g.setChannels(&pkg)
+// sortChannelsByName sort the channels by the name
+// NOTE: sorting makes the channel order always consistent when appending new channels
+func (g pkgGenerator) sortChannelsByName(pkg *registry.PackageManifest) {
 	sort.Slice(pkg.Channels, func(i int, j int) bool {
 		return pkg.Channels[i].Name < pkg.Channels[j].Name
 	})
+}
 
-	b, err := yaml.Marshal(pkg)
-	if err != nil {
-		return nil, err
+// validatePackageManifest will verify the PackageManifest by using the API
+// More info: https://github.com/operator-framework/api
+func validatePackageManifest(pkg *registry.PackageManifest) error {
+	if pkg == nil {
+		return fmt.Errorf("unable to check Package Manifest, verify its artefacts in the project")
```

Comment:
```suggestion
		return errors.New("generated package manifest is empty")
```
A valid package manifest will always be passed to `validatePackageManifest` given how it is called, so no user-facing error information is needed.